pyswmm/swmm5.py

```diff
@@ -356,6 +357,26 @@ def swmm_close(self):
         """
         solver.swmm_close()
         self.fileLoaded = False
+    
+    def swmm_save_hotstart(self, hotstart_filename):
```

Comment:
@karosc for this we should throw a version exception if we are <5.2.3 (4)?

Reply:
@bemcdonnell, I thought our intention was to [remove support for older swmm-toolkit versions](https://github.com/pyswmm/pyswmm/blob/5feb6e751f565fc135d1d8dcc9711a9c2f0eefba/setup.py#L56)

Do we want to still allow users to install older versions of swmm? That would also require us to build some exception handling in the updates swmm_stride function too since it will use the [C-code](https://github.com/pyswmm/Stormwater-Management-Model/blob/8266e4434bc9c23218bbf18fe57b0f3523d414ee/src/solver/swmm5.c#L506-L550) rather than [python](https://github.com/pyswmm/pyswmm/blob/22aed805ec435dce93d0ccc084041e380827e01d/pyswmm/swmm5.py#L313-L319)

Reply:
Didn't we open up a swmm-toolkit version picker for pyswmm's install dep? If we have the option to use different versions of swmm-toolkit, then we need to sieve the functions accordingly.  My memory in English is rapidly disappearing so forgive me if I forget something.

Reply:
For `swmm_stride` let's intro that in pyswmm-2.  I think these latest features could fly for pyswmm-1.

Reply:
@bemcdonnell, I reverted some changes and added a [new decorator function](https://github.com/pyswmm/pyswmm/pull/476/files#diff-55fb4989cead9787f7865311d38585fa532dfd8993098ea262825b1b3c23af35R17-R35) that raises an exception if swmm-toolkit is less than a specified version. This way we can easily mark verious api functions as only compatible with a given minimum swmm-toolkit version

Reply:
This might just be me being persnickety, but is there anyway we can cache that version control variable the first time it’s called? I have a feeling this could slow things down when we start adding more and more functions that leverage it.

Reply:
What I meant to say FIRST was, this is awesome! 😊😊😊

Reply:
I don't see a good way to do that other than overriding the instance methods in the object __init__. In other words, when creating a sim object, check object methods against swmm-toolkit version and replace them with an exception thrower if your version of swmm-toolkit is not compatible.

Reply:
You’re asserting the SWMM-toolkit package number not the SWMM number, if I understand this correctly. If we throw an exception, it pushes the SWMM version number. This should be fine. No calls into SWMM until the exception is thrown :-)

Reply:
What if we just monkey patch the low level solver swig interface at pyswmm import? That way we can just replace missing functionality in the C-API with python functionality, or an exception thrower once at runtime.